packages/compass-data-modeling/src/store/diagram.ts

```diff
@@ -738,6 +756,26 @@ export function changeFieldType(
     });
     if (!field) throw new Error('Field not found in schema');
     const to = getSchemaWithNewTypes(field.jsonSchema, newTypes);
+
+    track('Data Modeling Field Type Changed', {
+      source: 'side_panel',
+      // If the field had a single type, we report that, otherwise 'mixed'.
+      from: field.jsonSchema.bsonType
+        ? Array.isArray(field.jsonSchema.bsonType)
+          ? field.jsonSchema.bsonType.length === 1
+            ? field.jsonSchema.bsonType[0]
+            : 'mixed'
+          : field.jsonSchema.bsonType
+        : undefined,
+      to: to.bsonType
+        ? Array.isArray(to.bsonType)
+          ? to.bsonType.length === 1
+            ? to.bsonType[0]
+            : 'mixed'
+          : to.bsonType
+        : undefined,
+    });
```

Comment:
Is this computation specific to telemetry? It feels to me like this has to also happen elsewhere 🙂